crates/engine/src/operation/solve/solver/adapter.rs

```diff
@@ -171,3 +264,50 @@ impl<'a> OperationAdapter<'a> {
         (start..end).into()
     }
 }
+
+/// Generating a response key for an extra field. The only constraint is that it needs to be unique
+/// within the selection set we send to a subgraph. Afterwards we identify extra fields through
+/// their associated `SchemaFieldId` so the response key doesn't matter past deserialization.
+fn generate_new_extra_field_key(
+    response_keys: &mut ResponseKeys,
+    selection_set: &[SafeResponseKey],
+    definition: FieldDefinition<'_>,
```

Comment:
What's the difference between this function and the very similarly named/written one above?  Answer in a comment please.